### PR TITLE
ADX-794-activity-stream-pagination

### PR DESCRIPTION
**Pull request by Manoj-nathwani**
https://fjelltopp.atlassian.net/browse/ADX-794

image

## Related
- https://github.com/fjelltopp/adx_develop/pull/89
- https://github.com/fjelltopp/adx_deploy/pull/228
- https://github.com/ckan/ckan/discussions/6751

### CODE REVIEW

File: ckanext/unaids/theme/templates/package/activity.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      table_rows += '</td></tr>';
+    });
+    $('#ActivityStreamContainer tbody').html(table_rows);
+
+    // convert table into paginated bootstrapTable
+    $('#ActivityStreamContainer table').bootstrapTable({
+      classes: 'table table-borderless',
+      showHeader: false,
+      pagination: true,
+      pageSize: 5,
```
Comment (Manoj-nathwani) [inline ckanext/unaids/theme/templates/package/activity.html:59]: @tomeksabala - please confirm 5 is a good default

**Comment by Manoj-nathwani** [comment]
We also need to update the following ckan config:
https://ckan.readthedocs.io/en/2.9/maintaining/configuration.html#ckan-activity-list-limit

File: ckanext/unaids/theme/templates/package/activity.html (hunk 1 above)
Comment (tomeksabala) [reply ckanext/unaids/theme/templates/package/activity.html:59]: It should be bigger. I think that current number of items displayed in the activity list is ok. You could use that.

File: ckanext/unaids/theme/templates/package/activity.html [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    display: none;
+  }
+
+  #ActivityStreamContainer td {
+    border: none;
+    padding-bottom: 15px;
+  }
+</style>
+
+<script>
```
Comment (tomeksabala) [inline ckanext/unaids/theme/templates/package/activity.html:42]: The js code should be a separate js file configured in webassets and later requested with:
```
{% asset 'xxx' %}
```

File: ckanext/unaids/theme/templates/package/activity.html [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+<div id="ActivityStreamContainer">
+  <table class="activity">
+    <thead>
+      <tr>
+        <th></th>
+      </tr>
+    </thead>
+    <tbody></tbody>
+  </table>
+</div>
```
Comment (tomeksabala) [inline ckanext/unaids/theme/templates/package/activity.html:29]: the div id="ActivityStreamLoading" is the actual extension of this template.
You don't have to copy the whole file over. With help of ckan jinja extension mechanism you can do it in a nicer way (without core code duplication).
Take a look at: https://docs.ckan.org/en/2.9/theming/templates.html#extending-templates-with-ckan-extends
(leave the block `primary_content_inner` intact and add your pagination html separatly, sth like:
```
{% ckan_extends %}
<div id="ActivityStreamLoading">
[...]
```

**Review by tomeksabala — CHANGES_REQUESTED** [review]
I think that your branch is checkout from master again instead of development branch.

**Comment by Manoj-nathwani** [comment]
Shelving this for now as it’s actively being worked on https://github.com/ckan/ckan/pull/6754

**Pull request closed without merging**